src/celeritas/global/alongstep/AlongStepRZMapFieldMscAction.cc

```diff
@@ -0,0 +1,102 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/global/alongstep/AlongStepRZMapFieldMscAction.cc
+//---------------------------------------------------------------------------//
+#include "AlongStepRZMapFieldMscAction.hh"
+
+#include <utility>
+
+#include "corecel/Assert.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/Range.hh"
+#include "corecel/data/Ref.hh"
+#include "corecel/sys/Device.hh"
+#include "corecel/sys/MultiExceptionHandler.hh"
+#include "celeritas/em/FluctuationParams.hh"
+#include "celeritas/em/UrbanMscParams.hh"
+#include "celeritas/field/RZMapField.hh"
+#include "celeritas/global/CoreTrackData.hh"
+#include "celeritas/global/KernelContextException.hh"
+#include "celeritas/global/alongstep/AlongStepLauncher.hh"
+
+#include "AlongStepLauncher.hh"
+#include "detail/AlongStepRZMapFieldMsc.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the along-step action from input parameters.
+ */
+std::shared_ptr<AlongStepRZMapFieldMscAction>
+AlongStepRZMapFieldMscAction::from_params(ActionId id,
+                                          MaterialParams const& materials,
+                                          ParticleParams const& particles,
+                                          RZMapFieldInput const& field_input,
+                                          SPConstMsc const& msc,
+                                          bool eloss_fluctuation)
+{
+    SPConstFluctuations fluct;
+    if (eloss_fluctuation)
+    {
+        fluct = std::make_shared<FluctuationParams>(particles, materials);
+    }
```

Comment:
The `eloss_fluctuation` bool needs to be removed since you dereference `fluct_` later (i.e., it's required).

Reply:
Removed.